tests/integration/targets/aci_l3out_logical_interface_profile_eigrp_profile/tasks/main.yml

```diff
@@ -0,0 +1,188 @@
+      <<: *aci_info
+      domain: l3outintftest
+      domain_type: l3dom
+      state: present
+
+  - name: Create tenant
+    cisco.aci.aci_tenant: &tenant_present
+      <<: *tenant_absent
+      state: present
+
+  - name: Configure VRF
+    cisco.aci.aci_vrf: &vrf_present
+      <<: *tenant_present
+      vrf: l3outintftest
+
+  - name: Create L3Out
+    cisco.aci.aci_l3out:
+      <<: *vrf_present
+      l3out: l3outintftest
+      domain: l3outintftest
+      route_control: export
+
+  - name: Create node profile
+    cisco.aci.aci_l3out_logical_node_profile: &np_present
+      <<: *tenant_present
+      l3out: l3outintftest
+      node_profile: NODES
+
+  - name: Create interface profile
+    cisco.aci.aci_l3out_logical_interface_profile: &intf_present
+      <<: *np_present
+      interface_profile: INTFS
+
+  - name: Create eigrp interface policy
+    cisco.aci.aci_interface_policy_eigrp: &interface_policy_eigrp_present
+      <<: *tenant_present
+      eigrp_interface: ansible_eigrp_interface
+      description: eigrp interface policy for ansible_tenant tenant
+      bandwidth: 100000
+      control_state: [bfd, nexthop_self, passive, split_horizon]
+      delay: 10
+      delay_unit: picoseconds
+      hello_interval: 10
+      hold_interval: 30
+      state: present
+
+  # BEGIN WITH TESTS (ADD EIGRP profile TO INTERFACE PROFILE)
+  - name: Add eigrp profile to interface profile (check mode)
+    cisco.aci.aci_l3out_logical_interface_profile_eigrp_profile: &eigrpIfP_present
+      <<: *intf_present
+      eigrp_profile: default
+    check_mode: true
+    register: cm_add_eigrpIfP
+
+  - name: Add eigrp profile to interface profile (normal mode)
+    cisco.aci.aci_l3out_logical_interface_profile_eigrp_profile: *eigrpIfP_present
+    register: nm_add_eigrpIfP
+
+  - name: Verify add_eigrpIfP
+    ansible.builtin.assert:
+      that:
+      - cm_add_eigrpIfP is changed
+      - nm_add_eigrpIfP is changed
+      - cm_add_eigrpIfP.previous == nm_add_eigrpIfP.previous == []
+      - cm_add_eigrpIfP.proposed.eigrpIfP.attributes.name == 'default'
+      - nm_add_eigrpIfP.current[0].eigrpIfP.attributes.name == 'default'
+
+  - name: Add eigrp profile again to check idempotency
+    cisco.aci.aci_l3out_logical_interface_profile_eigrp_profile: *eigrpIfP_present
+    register: nm_add_eigrpIfP_again
+
+  - name: Verify add_eigrpIfP_again
+    ansible.builtin.assert:
+      that:
+      - nm_add_eigrpIfP_again is not changed
+
+  # UPDATE INTERFACE PROFILE EIGRP PROFILE
+  - name: Update interface profile eigrp profile
+    cisco.aci.aci_l3out_logical_interface_profile_eigrp_profile: &eigrpIfP_update
+      <<: *eigrpIfP_present
+      eigrp_policy: ansible_eigrp_interface
+    register: update_eigrpIfP
+
+  - name: Update interface profile eigrp profile again
+    cisco.aci.aci_l3out_logical_interface_profile_eigrp_profile:
+      <<: *eigrpIfP_update
+    register: update_eigrpIfP_again
+
+  - name: Verify update_eigrpIfP
```

Comment:
why are you not asserting the eigrp_policy? and also the other auth child is not asserted to be updated

Reply:
added asserts for the eigrp_policy, removed auth configs from the module so asserts for auth config are no longer needed.